### DISCUSSION TITLE
Questions related configuration

### DISCUSSION THREAD

**Discussion by r1cebank** [ricsanfre/pi-cluster#120] [Q&A] [answered]
Hi Ricardo,

I want to thank you for preparing this project and your detailed documentation on how to setup this k3s cluster. I am using your repo and adapted to my x86 vm clusters. Currently I am running into a few issues I am wondering if you can provide some insight on where to start debugging.

1. I have updated the url for the data source configuration in grafana to not include `svc.cluster.local` and just include the service and namespace names, it seems like with `svc.cluster.local` pods cannot resolve the host.
2. The grafana icon on linkerd-viz dashboard opens up a new page that shows invalid username or password, I've tried to tweak configurations like `grafana.url` in linkerd-viz and nothing seems to work.
3. Also using emojivoto, the temp traces is a bit weird and do not seem to be showing what you have shown in the document, but I wasn't able to locate any logs in elasticsearch for me to investigate further.

Do you know where should I start to debug the temp tracing issue and if the dns and gradana issue i am having is expected?

Thank You

**Comment by r1cebank** [comment]
I was able to resolve the issue 1 with the dns, it turns out the custom resolv.conf wasn't picked up during install and it was using a domain search on my local domain. For the temp trace of emojivoto, i couldn't able to reproduce what you were seeing, i only see traces ends with linkerd proxy
![image](https://user-images.githubusercontent.com/261671/231621709-c93053fe-677e-49e5-a70e-fec05f7ccd8a.png)

**Comment by ricsanfre** [comment, marked as answer]
Hi @r1cebank,

Thanks for your kind words and glad to hear that you are adapting this repo for working with your x86 VM cluster.

Regarding your issues/questions:

1. Happy to see that you already solve dns issue.

2. You are right, something should have changed in newer versions of linkerd/grafana breaking the integration. I have open issue #122 to track this error. I would be more than happy to accept any PR if you manage to solve it.

3. Are you using my ArgoCD pre-packaged applications?.  if you are using my repo with ArgoCD pre-packaged applications, there is one change in the emojivoto testing commands you need to do respect the ones included in the documentation. Documentation is not considering that tempo helm chart is a subchart. Tempo service name change when being part of another chart (tracing chart)

   When testing emojivoto application, command to configure emojivoto to emit spans to tempo (instructions step 2) need to be changed:

   ```
   kubectl -n emojivoto set env --all deploy OC_AGENT_HOST=tempo-distributor.tracing:55678
   ```
   must be changed to:
   ```
   kubectl -n emojivoto set env --all deploy OC_AGENT_HOST=tracing-tempo-distributor.tracing:55678
   ```
  
  Let me know if this is your case or you are facing any other issue.



Regards
Ricardo

**Reply by ricsanfre** [reply]
Issue #122 already solved. 

Adding removeHeaders option to BasicAuth Traefik Middleware is enough to solve the issue 2.

**Reply by r1cebank** [reply]
Thank you for the detailed instruction on how to fix the issue I am having, it all worked. I also opened up an PR to allow the external secret generated for elasticsearch to be used for the elasicsearch cluster.
